tst/EnergyPlus/unit/HVACControllers.unit.cc

```diff
@@ -133,16 +133,19 @@ TEST_F(EnergyPlusFixture, HVACControllers_ResetHumidityRatioCtrlVarType)
 
     GetSetPointManagerInputs(*state);
     // check specified control variable type is "HumidityRatio"
-    ASSERT_TRUE(compare_enums(SetPointManager::CtrlVarType::HumRat, state->dataSetPointManager->AllSetPtMgr(1).CtrlTypeMode));
+    ASSERT_EQ((int)HVAC::CtrlVarType::HumRat, (int)state->dataSetPointManager->spms(1)->ctrlVar);
```

Comment:
This is better than `compare_enums`. Just use this from now on.

Reply:
I disagree. IMHO The custom function is clearer, and doesn't require casting, so you don't have to remember how to do it properly. But that's not even the real point here.

The point is that compare_enums will not let you shoot yourself in the foot. You **cannot** compare different enums with it, it will not compile.

---

And I'd avoid C-like cast, so at least it should be a static_cast<int>, which gets verbose.

An alternative would be define the C++23 std::to_underlying: https://en.cppreference.com/w/cpp/utility/to_underlying

That's dead simple

```
// std::to_underlying only in C++23
template <class T>
inline constexpr typename std::underlying_type_t<T> to_underlying(T val) noexcept {
  return static_cast<typename std::underlying_type<T>::type>(val);
}
```

But it'd still won't ensure you use the right types

Demo: https://godbolt.org/z/57PYfrs1Y

Reply:
How about we figure a way to have  `EXPECT_ENUM_EQ` and a `EXPECT_ENUM_NE` macros.

You could use them like

```c++
    auto v1 = EnumClass::val1;
    EXPECT_ENUM_EQ(v1, EnumClass::val2);
    EXPECT_ENUM_NE(v1, EnumClass::val1);
```

And they'd format to 

```shell
/path/tofile.cpp:107: Failure
In comparing enums of type 'EnumClass', Expected equality of these values:
  v1
    Which is: 0
  EnumClass::val2
    Which is: 1

/path/tofile.cpp:110: Failure
In comparing enums of type 'EnumClass'
  Expected: (v1) != (EnumClass::val1)
  Actual: both are 0
[  FAILED  ] Enum.RegularEnum (0 ms)
```

Reply:
My beef is not with the error message, it's with the `compare_enums(X1, X2, false)` "gotcha", so I like the `EXPECT_ENUM_EQ`/`EXPECT_ENUM_NE` solution. We don't have to do it in this PR. I will revert these changes and we can do this separately. Thanks.

Reply:
I reverted.